### PR TITLE
Create ranobelib.py

### PR DESCRIPTION
**Pull request by needKVAS**
New source

### CODE REVIEW

**Comment by zGadli** [comment]
Tried to visit the website by copying the base_url. It gives me a 404 error. Please check if this is an issue or not.

**Comment by needKVAS** [comment]
Base_url works fine for me. I don't think it's a source file issue.

**Comment by zGadli** [comment]
@needKVAS It might be because the website can not be accessed from my country. Maybe it will work if i use a VPN.

**Comment by zGadli** [comment]
@needKVAS I tried using proxies and VPN yet I still can't access this website. It shows 404 error. Somebody else should confirm if they can access the source website,

**Comment by needKVAS** [comment]
@zGadli Yes, the website is really not available for most countries: [https://check-host.net/check-report/112e7396k5a2](https://check-host.net/check-report/112e7396k5a2)

**Comment by zGadli** [comment]
@needKVAS Oh! Thanks for the info.

File: sources/ru/ranobelib.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        for script in soup.find_all("script"):
+            json_var = "window.__DATA__ = "
+            text = script.text.strip()
+            if not text or not text.startswith(json_var):
+                continue
+            text = text[len(json_var) : text.find("window._SITE_COLOR_")].strip(
+                ";\t\n "
+            )
+            content = json.loads(text)
```
Comment (dipu-bd) [inline sources/ru/ranobelib.py:32]: you could use a `break` after this

File: sources/ru/ranobelib.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.novel_title = content["manga"]["rusName"]
+        logger.info("Novel title: %s", self.novel_title)
+
+        self.novel_cover = self.absolute_url(
+            soup.select_one(".media-sidebar__cover > img:nth-child(1)")["src"]
+        )
+        logger.info("Novel cover: %s", self.novel_cover)
+
+        self.novel_author = soup.select_one(
+            "div.media-info-list__item:nth-child(6) > div:nth-child(2) > a:nth-child(1)"
```
Comment (dipu-bd) [inline sources/ru/ranobelib.py:43]: handle the case where this tag might not exists. it should not throw error the crawler if novel author tag is not found. the author is optional.

**Comment by needKVAS** [comment]
@dipu-bd  Thanks for the review. Fixed.

**Pull request merged**